src/container/AddToken.container.tsx

```diff
@@ -57,12 +67,25 @@ export const AddTokenContainer = observer(({navigation}: IProps) => {
       return;
     }
 
-    root.node.addToken(source, name, key);
+    if (isGitlab) {
+      if(!baseURL){
+        root.ui.addToast({
+          text: 'Please add a base URL',
+          type: 'error',
+        });
+        return;
+      }
+
+      root.node.addToken(source, name, key, baseURL, visibility);
```

Comment:
maybe trim() and sanitize the url before adding it? at least checking that it is a valid URL

Reply:
thought about it but didn't want to add an external dependency. I will add regex based validation (`baseURL` can be a domain or an IP address)

Reply:
Please use: https://github.com/ospfranco/link-preview-js/blob/main/constants.ts

Reply:
Perfect 👍